parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -160,9 +161,10 @@ public static class WriteBuilder implements InternalData.WriteBuilder {
     private final Map<String, String> metadata = Maps.newLinkedHashMap();
     private final Map<String, String> config = Maps.newLinkedHashMap();
     private Schema schema = null;
+    private BiFunction<Integer, String, Type> variantShreddingFunc = null;
```

Comment:
I feel less and less comfortable about using functional references like this in public interfaces.  This really makes maintainability much harder because we're not documenting what this represents.  If we're going have a reference like this, then we should just make an wrapper interface (e.g. `VariantShreddingFunction`) that documents the parameters/return and allows for better navigation of usage.  

This is particularly prevalent in parquet and avro, but it even here we have three different functional references that we're passing around.  If it were internal, I'd be fine, but these are public apis so we should do better.

Reply:
You're right. I think this should have been a private constructor and this should only be available through the `ParquetSchemaUtil` method that documents how it is used. I'll update it.

Reply:
I updated the config method to document this and explain what is happening.

Reply:
Updated to use `VariantShreddingFunction`.